addons/mail/controllers/discuss/gif.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import requests
+import urllib.parse
+from odoo.http import request, route, Controller
+
+
+class DiscussGifController(Controller):
+
+    def _request_gifs(self, url):
+        try:
+            response = requests.get(url, timeout=3)
+        except requests.exceptions.RequestException:
+            return False
+        if response.status_code != requests.codes.ok:
+            return False
+        return response
+
+    @route("/discuss/gif/search", type="json", auth="public", cors="*")
+    def search(self, search_term, locale="en", country="US", position=False):
+        config = request.env["ir.config_parameter"].sudo().tenor_query_params()
+        url = f"https://tenor.googleapis.com/v2/search?q={urllib.parse.quote_plus(search_term)}&key={config.get('api_key')}&client_key={config.get('client_key')}&limit={config.get('tenor_gif_limit')}&contentfilter={config.get('content_filter')}&locale={urllib.parse.quote_plus(locale)}&country={urllib.parse.quote_plus(country)}&media_filter=tinygif"
```

Comment:
that's not easy to read, maybe reformat on multiple lines, using `format`instead?

Reply:
Using `werkzeug.urls.url_encode` clean this.